lib/src/kbasesearchengine/parse/KeywordParser.java

```diff
@@ -33,6 +34,19 @@ public static ParsedObject extractKeywords(
             final ObjectLookupProvider lookup,
             final List<GUID> objectRefPath)
             throws IOException, ObjectParseException, IndexingException, InterruptedException {
+
+        // check pre-conditons
+        if(json == null) throw new IllegalArgumentException("json is a required parameter");
+        if(indexingRules == null) throw new IllegalArgumentException("indexingRules is a required parameter");
+
+        for(IndexingRules rule: indexingRules) {
+            try {
+                rule.validate();
```

Comment:
I assuming we'll do the validation when we build the rules at some point rather than every time this method is called

Reply:
I have added a validation check at the building stage as well.